### ISSUE TITLE
Enormous gif recording file size

### ISSUE THREAD

**Issue by MarvinJWendt** [ShareX/ShareX#5183]
The option to record a `gif` should have a setting to compress the generated gif and change the result. Since I have a 4k screen, recorded GIFs with less than 20 seconds are already several hundred MB in size. Unfortunately this makes the function very unuseful for me, because I want to create the GIFs to upload them in forums as comments. Most of these forums, like the issues here on GitHub, are limited to a maximum of 10MB.

**Comment by skeddles** [comment]
Just want to say that it needs to shrink the gif's resolution and framerate too, not just compress it. GIFs are already compressed, and even a single frame of a 4k gif would be huge. I think it would be best if you could set a target max filesize.

**Comment by MarvinJWendt** [comment]
Yes, I also thought about that. Probably an option for both would be nice. Shrinking the resolution would still make the gifs way too large to share on social media. Most gif generators shorten the gif to a color palette. ffmpeg can do this, but it doesn't by default. A setting for this would be nice :)

**Comment by Neoony** [comment]
Just to mention
Usually I am recording a normal video with Radeon and then playing that video in a player and making a gif from it.
That way you just scale the player window, to whatever size you want and then upload to imgur which supports 200mb gifs

But yeah, could use better controls for gifs definitely

**Comment by MarvinJWendt** [comment]
@Jaex what do you think about it?

**Comment by MarvinJWendt** [comment]
Not stale

**Comment by MarvinJWendt** [comment]
Still not stale

**Comment by Evo-c** [comment]
Damn it's been years and this still isn't solved? I have my gif FPS set to 5 yet a 20 second video of 720p with barely much moving still has a file size of 30mb+

**Comment by MarvinJWendt** [comment]
Sadly, I think this project is near death. Issues don't get resolved, not even feature-breaking ones like: #6570 (also by me). I have given up and as much as I like ShareX and promoted it in the past, I am looking for alternatives now.

**Comment by MarvinJWendt** [comment]
Oh and just for the record, this issue here could be fixed in like 10 minutes. ShareX already uses FFmpeg, which is capable to shrink GIFs...

**Comment by wolfred24** [comment]
I need this function also :(

**Comment by Ione15** [comment]
Either use Gifski or use a video instead! (alternatively check out webp avif or jxl animated)

**Comment by MarvinJWendt** [comment]
> Either use Gifski or use a video instead! (alternatively check out webp avif or jxl animated)

That's not a solution, that's an alternative. As already mentioned, GIFs are great for uploading in forums like GitHub issues, videos are not. Obviously you can shrink the GIF afterwards, but that's an unnecessary step that shouldn't be required.

**Comment by brunolm** [comment]
Any updates? Is there a way to purposely make the quality bad so it gets smaller? Or resize it automatically?

**Comment by 3meraldK** [comment]
I support this idea wholeheartedly :P

**Comment by Ione15** [comment]
> > Either use Gifski or use a video instead! (alternatively check out webp avif or jxl animated)
> 
> That's not a solution, that's an alternative. As already mentioned, GIFs are great for uploading in forums like GitHub issues, videos are not. Obviously you can shrink the GIF afterwards, but that's an unnecessary step that shouldn't be required.

Fair, however are an built in feature for ShareX, as such you could just set the Input filter to .gif and apply whatever compression you want yourself. Or even use gifski with that.

**Comment by jack27121** [comment]
Whenever i record a gif with share x, i convert it into a gif (again) using File converter. And it becomes atleast half as big
https://file-converter.io/
Something similar would be great to have built in. File converter also has an option to shrink a gif when converting

**Comment by Fazyfaz** [comment]
Having the same problem. My 10 seconds GIF is 142 MB. Why is it so big? I don't get it

**Comment by Iggoor7** [comment]
I love the ability to make gifs with just one button! But my 2k screen is ruining it! Usually, you make gift to quickly show something to someone, and you don't need t high quality. It would be amazing if there was a setting to compress your gift to certain resolution! 

I want 720p gifs!

**Comment by brunolm** [comment]
tl;dr; GIFs suck

---

# Tests

This is an external post-processing workaround using `ffmpeg`.

I tested with a 3202x1304, ~29.75s screen recording.

Original MP4: 57.72 MiB / 60.52 MB

## Option 1: optimize lower resolution

Reduces resolution to 1920x782.

Result:

- MP4: 3.92 MiB / 4.11 MB
- GIF, 8 fps: 56.38 MiB / 59.12 MB

```powershell
ffmpeg -y -i .\test2.mp4 -map 0:v:0 -map 0:a? -vf scale=1920:-2:flags=lanczos -c:v libx264 -preset slow -crf 28 -pix_fmt yuv420p -c:a aac -b:a
128k -movflags +faststart .\test2-optimized.mp4
```

Convert to GIF:

```powershell
ffmpeg -y -i .\test2-optimized.mp4 -filter_complex "[0:v]fps=8,split[a][b];[a]palettegen[p];[b][p]paletteuse=dither=bayer" .\test2-optimized-
8fps.gif
```

## Option 2: optimize same resolution

Keeps the original 3202x1304 resolution.

Result:

- MP4: 8.02 MiB / 8.41 MB
- GIF, 8 fps: 145.76 MiB / 152.84 MB

```powershell
ffmpeg -y -i .\test2.mp4 -map 0:v:0 -map 0:a? -c:v libx264 -preset slow -crf 28 -pix_fmt yuv420p -c:a aac -b:a 128k -movflags +faststart .\test2-
optimized-same-res.mp4
```

Convert to GIF:

```powershell
ffmpeg -y -i .\test2-optimized-same-res.mp4 -filter_complex "[0:v]fps=8,split[a][b];[a]palettegen[p];[b][p]paletteuse=dither=bayer" .\test2-
optimized-same-res-8fps.gif
```

## Notes

File size depends a lot on motion, colors, and screen resolution, so these numbers are not guaranteed for every recording.


## Tools used

- [PowerToys](https://learn.microsoft.com/windows/powertoys/) > ZoomIt > Record
- [ffmpeg](https://ffmpeg.org/download.html)

## Final recommendation

For the smallest practical upload size:

1. Record/export as MP4 instead of GIF.
2. (optional) Downscale recordings.
3. Re-encode with H.264 using CRF-based compression
